### PR TITLE
Add next configuration as an option for custom servers

### PR DESCRIPTION
**Pull request by reel**
Instead of having a `next.config.js` file, we can dynamically add configuration from the custom server.

The use case would be to support many different webpack configuration with a `minimist` flag.

I can add an example if wanted.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -24,14 +24,14 @@ const internalPrefixes = [
...
+  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = {} } = {}) {
     this.dir = resolve(dir)
     this.dev = dev
     this.quiet = quiet
     this.router = new Router()
-    this.hotReloader = dev ? new HotReloader(this.dir, { quiet }) : null
+    this.hotReloader = dev ? new HotReloader(this.dir, { quiet, conf }) : null
     this.http = null
-    this.config = getConfig(this.dir)
+    this.config = getConfig(this.dir, conf)
```
Comment (arunoda) [inline server/index.js]: If we've a config via the option, it shouldn't read from the filesystem. Change the code to reflect that.

And make sure to get default configurations. May be we need to add method called `withDefaults` to `server/config.js`

**Review by arunoda — CHANGES_REQUESTED** [review]
The idea looks great to me.
Could you work on this change?

File: server/index.js (hunk 1 above)
Comment (reel) [inline server/index.js]: Well, for my use case, I still want to be able to read from filesystem to overwrite default & custom server.

That's why I have this order in Object.assign

Your thoughts?

Comment (reel) [inline server/index.js]: @arunoda Is this OK?

**Review by arunoda — APPROVED** [review]

**Comment by arunoda** [comment]
@reel everything looks great to me. Let's wait for tests.

**Comment by arunoda** [comment]
Actually the error is not our fault.
See: https://github.com/zeit/styled-jsx/pull/217

So. I'm taking this in.

**Comment by reel** [comment]
Thank's @arunoda 

Here's a peak at what we can do now: https://github.com/usehenri/henri/blob/master/packages/view/conf.js

**Pull request merged**